pkg/lib/operatorlister/clusterserviceversion.go

```diff
@@ -0,0 +1,95 @@
+package operatorlister
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	listers "github.com/operator-framework/operator-lifecycle-manager/pkg/api/client/listers/operators/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type UnionClusterServiceVersionLister struct {
+	csvListers map[string]listers.ClusterServiceVersionLister
+	csvLock    sync.RWMutex
+}
+
+// List lists all ClusterServiceVersions in the indexer.
+func (ucl *UnionClusterServiceVersionLister) List(selector labels.Selector) (ret []*v1alpha1.ClusterServiceVersion, err error) {
```

Comment:
Is this implementing some required interface? It seems like these could just be pure functions...

Reply:
This is implementing `v1alpha1.ClusterServiceVersionLister`.